rclcpp_components/src/node_main.cpp.in

```diff
@@ -41,13 +42,15 @@ int main(int argc, char * argv[])
   RCLCPP_DEBUG(logger, "Load library %s", library_name.c_str());
   auto loader = std::make_unique<class_loader::ClassLoader>(library_name);
   auto classes = loader->getAvailableClasses<rclcpp_components::NodeFactory>();
-  for (const auto & clazz : classes) {
-    std::string name = clazz.c_str();
-    if (name.compare(class_name) == 0) {
-      RCLCPP_DEBUG(logger, "Instantiate class %s", clazz.c_str());
+
+  if (std::find(
+    classes.begin(),
+    classes.end(),
+    class_name) != classes.end()) {
```

Comment:
Personally, I'd prefer to have less nesting.  So I'd invert the check and return the error early if we didn't find the class:

```suggestion
  if (std::find(
    classes.begin(),
    classes.end(),
    class_name) == classes.end())
  {
      RCLCPP_ERROR(
        logger,
        "Class %s not found in library %s",
        class_name.c_str(),
        library_name.c_str());
      return 1;
  }
```

Reply:
(and then unindent the loading code below)

Reply:
Done